CONTRIBUTING.md

```diff
@@ -740,3 +740,23 @@ Names of files and directories should be in lowercase, with dashes between words
 
   As an exception, an explicit conditional expression with null can be used when fixing a important bug without triggering a mass rebuild.
   If this is done a follow up pull request _should_ be created to change the code to `lib.optional(s)`.
+
+#### Use of `with`
+
+`with` is a somewhat controversial feature of nix syntax, it can reduce repition, but easily can make code unreadable or buggy if used carelessly.
```

Comment:
perhaps link to the `nix.dev` documentation here?

Reply:
i considered it, but honestly, i don't think very highly of that guide, as it frankly lacks a lot of nuance, and feels like it might as well call `with` a deprecated feature.  it doesn't provide any examples of times it would be ok or even encouraged to use.

Reply:
That guide can be improved too: https://github.com/NixOS/nix.dev